File: data_structures/ll_stack/python/linkedListStack.py
class Node:
    def __init__(self, val = None, nextNode = None):
        self.val = val
        self.nextNode = nextNode


class LinkedListStack:
    head = None
    length = 0

    def push(self, val):
        self.length += 1
        new_node = Node(val=val)
        if self.head is None:
            self.head = new_node
        else:
            new_node.nextNode=self.head
            self.head = new_node
        
    def pop(self):
        if self.length == 0:
            print("Stack is empty")
        else:
            self.length -= 1
            print(self.head.val)
            if self.length ==1:
                self.head = None
            else:
                self.head=self.head.nextNode

    def peek(self):
        if self.head == None:
            print("Queue is empty")
        else:
            print(self.head.val)

if __name__ == "__main__":
    stack = LinkedListStack()
    stack.peek()
    stack.push(5)
    stack.peek()
    stack.push(4)
    stack.push(3)
    stack.peek()
    stack.pop()
    stack.peek()

File: data_structures/ll_stack/java/linkedListStack.java
import java.util.LinkedList;

class LinkedListStack
{
    Node head;
    int length = 0;

    public void peek()
    {
        if(this.length == 0)
        {
            System.out.println("Stack is empty");
        }
        else
        {
            System.out.println(this.head.val);
        }
    }

    public void pop()
    {
        if(this.length == 0)
        {
            System.out.println("Stack is empty");
        }
        else if(this.length == 1)
        {
            this.length--;
            this.head = null;
        }
        else {
            System.out.println(this.head.val);
            this.head = this.head.prev;
        }
    }

    public void push(int val)
    {
        length++;
        Node new_node = new Node(val);

        if(this.length == 1)
        {
            this.head = new_node;
        }
        else
        {
            new_node.prev = this.head;
            this.head = new_node;
        }
    }

    public static void main(String[] args)
    {
        LinkedListStack stack = new LinkedListStack();
        stack.peek();
        stack.push(1);
        stack.peek();
        stack.push(2);
        stack.push(3);
        stack.push(4);
        stack.peek();
        stack.pop();
        stack.pop();
        stack.peek();

    }
}